src/entities.ts

```diff
@@ -0,0 +1,415 @@
+      ),
+      remaining: SchemaType.Optional(SchemaType.Number()),
+      orderNumber: SchemaType.Optional(SchemaType.String()),
+      purchaseCost: SchemaType.Optional(
+        SchemaType.Union([SchemaType.String(), SchemaType.Null()]),
+      ),
+      purchaseDate: SchemaType.Optional(SchemaType.Number()),
+      qty: SchemaType.Optional(SchemaType.Number()),
+      notes: SchemaType.Optional(SchemaType.Array(SchemaType.String())),
+      userCanCheckout: SchemaType.Boolean(),
+      'availableActions.checkout': SchemaType.Optional(
+        SchemaType.Boolean({
+          deprecated: true,
+          description: 'Please use availableActionsCheckout instead.',
+        }),
+      ),
+      availableActionsCheckout: SchemaType.Optional(SchemaType.Boolean()),
+      'availableActions.checkin': SchemaType.Optional(
+        SchemaType.Boolean({
+          deprecated: true,
+          description: 'Please use availableActionsCheckin instead.',
+        }),
+      ),
+      availableActionsCheckin: SchemaType.Optional(SchemaType.Boolean()),
+      'availableActions.update': SchemaType.Optional(
+        SchemaType.Boolean({
+          deprecated: true,
+          description: 'Please use availableActionsUpdate instead.',
+        }),
+      ),
+      availableActionsUpdate: SchemaType.Optional(SchemaType.Boolean()),
+      'availableActions.delete': SchemaType.Optional(
+        SchemaType.Boolean({
+          deprecated: true,
+          description: 'Please use availableActionsDelete instead.',
+        }),
+      ),
+      availableActionsDelete: SchemaType.Optional(SchemaType.Boolean()),
+    }),
+  });
+
+export const [LicenseEntityMetadata, createLicenseAssignEntity] =
+  createEntityMetadata({
+    resourceName: 'License',
+    _class: ['Application'],
+    _type: createEntityType('licensed_application'),
+    description: 'Snipe-IT License',
+    schema: SchemaType.Object({
+      licenseId: SchemaType.Optional(SchemaType.Number()),
+      company: SchemaType.Optional(SchemaType.String()),
+      'manufacturer.id': SchemaType.Optional(SchemaType.Number()),
+      manufacturerId: SchemaType.Optional(SchemaType.Number()),
+      'manufacturer.name': SchemaType.Optional(SchemaType.String()),
+      manufacturerName: SchemaType.Optional(SchemaType.String()),
+      productKey: SchemaType.Optional(SchemaType.String()),
+      orderNumber: SchemaType.Optional(SchemaType.String()),
+      purchaseOrder: SchemaType.Optional(SchemaType.String()),
+      purchaseDate: SchemaType.Optional(SchemaType.Number()),
+      terminationDate: SchemaType.Optional(SchemaType.Number()),
+      depreciation: SchemaType.Optional(SchemaType.String()),
+      purchaseCost: SchemaType.Optional(
+        SchemaType.Union([SchemaType.String(), SchemaType.Null()]),
+      ),
+      notes: SchemaType.Optional(SchemaType.Array(SchemaType.String())),
+      expirationDate: SchemaType.Optional(SchemaType.Number()),
```

Comment:
```suggestion
      purchasedOn: SchemaType.Optional(SchemaType.Number()),
      terminatedOn: SchemaType.Optional(SchemaType.Number()),
      depreciation: SchemaType.Optional(SchemaType.String()),
      purchaseCost: SchemaType.Optional(
        SchemaType.Union([SchemaType.String(), SchemaType.Null()]),
      ),
      notes: SchemaType.Optional(SchemaType.Array(SchemaType.String())),
      expiresOn: SchemaType.Optional(SchemaType.Number()),
```
Did we approve the date RFC?

Suggestions here are for perhaps a future iteration since these changes are breaking. Or we could deprecate and add? Thoughts?

Reply:
> Or we could deprecate and add

I'm definitely in favor of this approach

Reply:
We haven't approved any official Date changes